### PR TITLE
bugfix alarm time span match in silence and notice

### PR DESCRIPTION
**Pull request by tomsun28**
## What's changed?

bugfix alarm time span match in silence and notice

## Checklist

- [x]  I hereby agree to the terms of the [HertzBeat CLA](https://gist.github.com/tomsun28/511c04e7643901cb550bb6ecc75a661b)
- [ ]  I have read the [Contributing Guide](https://hertzbeat.com/docs/others/contributing/)
- [ ]  I have written the necessary doc or comment.
- [ ]  I have added the necessary unit tests and all cases have passed.

### CODE REVIEW

File: alerter/src/main/java/org/dromara/hertzbeat/alert/reduce/AlarmSilenceReduce.java [hunk 1]
```diff
@@ -72,25 +72,28 @@ public boolean filterSilence(Alert alert) {
...
 							LocalTime nowTime = nowDate.toLocalTime();
-							
-							if (nowTime.isAfter(alertSilence.getPeriodStart().toLocalTime())
-									&& nowTime.isBefore(alertSilence.getPeriodEnd().toLocalTime())) {
+							boolean startMatch = alertSilence.getPeriodStart() == null || 
+									nowTime.isAfter(alertSilence.getPeriodStart().toLocalTime());
+							boolean endMatch = alertSilence.getPeriodEnd() == null ||
+									nowTime.isBefore(alertSilence.getPeriodEnd().toLocalTime());
+							if (startMatch && endMatch) {
 								int times = alertSilence.getTimes() == null ? 0 : alertSilence.getTimes();
```
Comment (zqr10159) [inline alerter/src/main/java/org/dromara/hertzbeat/alert/reduce/AlarmSilenceReduce.java:97]: int times = Optional.ofNullable(alertSilence.getTimes()).orElse(0);

**Review by zqr10159 — COMMENTED** [review]
这两个地方可不可以用Optional

**Comment by hertzbeat** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
> Can Optional be used in these two places?

Received Updated

**Review by zqr10159 — APPROVED** [review]
👍👍👍

**Pull request merged**